### PR TITLE
Add Mgmt Mongo Cluster

### PR DESCRIPTION
**Pull request by v-jiaodi**
Pass

### CODE REVIEW

**Comment by kazrael2119** [comment]
operation name change: beginXXX -> xxx
operations have a new parameter ```subscriptionId```
Interface XXXOptionalParams no longer has parameter resumeFrom
client name chanaged
removed enum
removed XXXResponse

**Comment by kazrael2119** [comment]
different promiss model between hlc and modular:
https://github.com/Azure/autorest.typescript/pull/2581/files#diff-4b604745b828f172967e4f4fce6011caf7ee649de8665f5167bdf9c65358f74fR36
 
https://github.com/kazrael2119/azure-sdk-for-js/blob/mongcluster-re/sdk/mongocluster/arm-mongocluster/src/operationsInterfaces/firewallRules.ts#L63-L66

tsp link:https://github.com/Azure/autorest.typescript/blob/eb4289ea18401048dd5b82620868321e2015aa57/packages/typespec-test/test/DocumentDB.MongoCluster.Management/spec/FirewallRule.tsp#L39-L47

File: packages/typespec-test/test/DocumentDB.MongoCluster.Management/generated/typespec-ts/src/api/operations/index.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { PagedOperation, Operation } from "../../models/models.js";
```
Comment (MaryGao) [inline packages/typespec-test/test/DocumentDB.MongoCluster.Management/generated/typespec-ts/src/api/operations/index.ts:4]: Should not export to top-level index.

File: packages/typespec-test/test/DocumentDB.MongoCluster.Management/generated/typespec-ts/src/api/documentDBContext.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+export interface DocumentDBClientOptions extends ClientOptions {
+  /** The API version to use for this operation. */
+  apiVersion?: string;
+}
+
+export { DocumentDBContext } from "../rest/index.js";
+
+/** The Microsoft Azure management API provides create, read, update, and delete functionality for Azure Cosmos DB for MongoDB vCore resources including clusters and firewall rules. */
+export function createDocumentDB(
+  credential: TokenCredential,
```
Comment (MaryGao) [inline packages/typespec-test/test/DocumentDB.MongoCluster.Management/generated/typespec-ts/src/api/documentDBContext.ts:18]: no apiVersion parameter.

**Comment by qiaozha** [comment]
close it as no need for this now.

**Pull request closed without merging**